src/map/skill.c

```diff
@@ -20323,26 +20323,32 @@ static int skill_poisoningweapon(struct map_session_data *sd, int nameid)
 		clif->skill_fail(sd, GC_POISONINGWEAPON, USESKILL_FAIL_LEVEL, 0, 0);
 		return 0;
 	}
-	switch( nameid )
-	{ // t_lv used to take duration from skill->get_time2
-		case ITEMID_POISON_PARALYSIS:     type = SC_PARALYSE;      break;
-		case ITEMID_POISON_FEVER:         type = SC_PYREXIA;       break;
-		case ITEMID_POISON_CONTAMINATION: type = SC_DEATHHURT;     break;
-		case ITEMID_POISON_LEECH:         type = SC_LEECHESEND;    break;
-		case ITEMID_POISON_FATIGUE:       type = SC_VENOMBLEED;    break;
-		case ITEMID_POISON_NUMB:          type = SC_TOXIN;         break;
-		case ITEMID_POISON_LAUGHING:      type = SC_MAGICMUSHROOM; break;
-		case ITEMID_POISON_OBLIVION:      type = SC_OBLIVIONCURSE; break;
-		default:
-			clif->skill_fail(sd, GC_POISONINGWEAPON, USESKILL_FAIL_LEVEL, 0, 0);
-			return 0;
+
+	int msg = 0;
+	switch (nameid) {
+	case ITEMID_POISON_PARALYSIS:     msg = MSG_PARALYZE;      type = SC_PARALYSE;      break;
+	case ITEMID_POISON_FEVER:         msg = MSG_PHYREXIA;      type = SC_PYREXIA;       break;
+	case ITEMID_POISON_CONTAMINATION: msg = MSG_DEATHHURT;     type = SC_DEATHHURT;     break;
+	case ITEMID_POISON_LEECH:         msg = MSG_RICHEND;       type = SC_LEECHESEND;    break;
+	case ITEMID_POISON_FATIGUE:       msg = MSG_VENOMBLEED;    type = SC_VENOMBLEED;    break;
+	case ITEMID_POISON_NUMB:          msg = MSG_TOXIN;         type = SC_TOXIN;         break;
+	case ITEMID_POISON_LAUGHING:      msg = MSG_MAGICMUSHROOM; type = SC_MAGICMUSHROOM; break;
+	case ITEMID_POISON_OBLIVION:      msg = MSG_OBLIANCURSE;   type = SC_OBLIVIONCURSE; break;
```

Comment:
I was conflicted between following the standard and putting each assignment in a line or keeping it like that...

It does look easier to read when adding everything in a single line. But let me know your thoughts and I can change.

If there is an even better way than the switch, I am also open to suggestions :)

Reply:
I agree that this tabular form makes it easier on the eye (and it's short enough so that it's feasible to keep it aligned). The day we switch to clang-format to auto-format the code, we can put ignore directives around blocks similar to this so that their custom format is preserved.